app/src/main/java/com/rkm/tasky/repository/implementation/TaskyTaskRepositoryImpl.kt

```diff
@@ -0,0 +1,82 @@
+package com.rkm.tasky.repository.implementation
+
+import com.rkm.tasky.database.dao.SyncDao
+import com.rkm.tasky.database.dao.TaskDao
+import com.rkm.tasky.database.model.SyncEntity
+import com.rkm.tasky.database.model.SyncItemType
+import com.rkm.tasky.database.model.SyncUserAction
+import com.rkm.tasky.di.IoDispatcher
+import com.rkm.tasky.feature.task.model.Task
+import com.rkm.tasky.network.datasource.TaskyTaskRemoteDataSource
+import com.rkm.tasky.network.util.NetworkError
+import com.rkm.tasky.network.util.safeCall
+import com.rkm.tasky.repository.abstraction.TaskyTaskRepository
+import com.rkm.tasky.repository.mapper.asTask
+import com.rkm.tasky.repository.mapper.asTaskEntity
+import com.rkm.tasky.repository.mapper.asTaskyRequest
+import com.rkm.tasky.util.result.EmptyResult
+import com.rkm.tasky.util.result.Result
+import com.rkm.tasky.util.result.onFailure
+import kotlinx.coroutines.CoroutineDispatcher
+import kotlinx.coroutines.withContext
+import javax.inject.Inject
+
+class TaskyTaskRepositoryImpl @Inject constructor(
+    private val remoteDataSource: TaskyTaskRemoteDataSource,
+    private val localDataSource: TaskDao,
+    private val syncDataSource: SyncDao,
+    @IoDispatcher private val  dispatcher: CoroutineDispatcher
+): TaskyTaskRepository {
+    override suspend fun getTask(id: String): Result<Task?, NetworkError.APIError> = withContext(dispatcher){
+        val results = safeCall { remoteDataSource.getTask(id) }
+        results.onFailure {
+            return@withContext Result.Success(localDataSource.getTaskById(id)?.asTask())
+        }
+        return@withContext Result.Success((results as Result.Success).data.asTask())
+    }
```

Comment:
Considering this is an offline-first app, wouldn’t it make sense to fetch straight from the local DB? This would also reduce the number of API calls your app makes. You can also use the “/agenda” endpoint to get all the agenda items for a particular day.